internal/scheduler/legacyscheduler.go

```diff
@@ -791,6 +954,204 @@ func NewLegacyScheduler(
 	}, nil
 }
 
+func Reschedule(
+	ctx context.Context,
+	jobRepo JobRepository,
+	constraints SchedulingConstraints,
+	config configuration.SchedulingConfig,
+	nodeDb *NodeDb,
+	priorityFactorByQueue map[string]float64,
+	initialResourcesByQueueAndPriority map[string]schedulerobjects.QuantityByPriorityAndResourceType,
+	nodeFairShareEvictionProbability float64,
+	nodeOversubscriptionEvictionProbability float64,
+) ([]LegacySchedulerJob, []LegacySchedulerJob, map[string]*schedulerobjects.Node, error) {
```

Comment:
at the point we're returning a tuple of 4 objects I'd be tempted to make a wrapper struct.
I'd also put some doc on the fucntion to say what it does

Reply:
Agreed. Something for a later PR I think. I'd like to rework this whole file a bit.